x/lightclient/keeper/grpc_query_chain_state.go

```diff
@@ -0,0 +1,52 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/zeta-chain/zetacore/x/lightclient/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ChainStateAll queries all chain statess
+func (k Keeper) ChainStateAll(c context.Context, req *types.QueryAllChainStateRequest) (*types.QueryAllChainStateResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+	store := ctx.KVStore(k.storeKey)
+	chainStateStore := prefix.NewStore(store, types.KeyPrefix(types.ChainStateKey))
+
+	var chainStates []types.ChainState
+	pageRes, err := query.Paginate(chainStateStore, req.Pagination, func(key []byte, value []byte) error {
```

Comment:
NIT : same as block headers , we can add a function `GetChainStatePaginated`

Reply:
1ee85715